kernel/component-manager/src/main/java/org/sakaiproject/util/SakaiApplicationContext.java

```diff
@@ -34,7 +34,7 @@
  * PostProcessor objects (e.g., SakaiProperties) a chance to do their work,
  * and load a few central components before the rest.
  */
-public class SakaiApplicationContext extends GenericApplicationContext {
+public class SakaiApplicationContext extends GenericWebApplicationContext {
```

Comment:
This probably shouldn't be a GenericWebApplicationContext, as the SakaiApplicationContext is not associated with a webapp.

Reply:
It is associated with the Spring Boot configuration via the App.java file:

ConfigurableApplicationContext sakaiApplicationContext = ((SpringCompMgr) ComponentManager.getInstance()).getApplicationContext();
SpringApplicationBuilder appBuilder = application.parent(sakaiApplicationContext).bannerMode(Banner.Mode.OFF).sources(App.class);

It was the only way I found to solve this:

Caused by: java.lang.IllegalArgumentException: Unable to call initializer. Object of class [org.sakaiproject.util.SakaiApplicationContext] must be an instance of interface org.springframework.web.context.ConfigurableWebApplicationContext

What do you think? Thanks